src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -2702,7 +2702,8 @@ private bool CheckInArgTypes(TexlNode left, TexlNode right)
                     return false;
                 }
 
-                if (_txb.Document.Properties.EnabledFeatures.IsEnhancedDelegationEnabled && typeLeft.IsTable)
+                if ((_txb.Document == null || _txb.Document.Properties.EnabledFeatures.IsEnhancedDelegationEnabled)
```

Comment:
>Document [](http://example.com/codeflow?start=26&length=8)

We use _txb.Document all over... I wonder if there are other places we could nullref...

_txb.Document just pulls from the resolver. 

Engine/Recalc engine produce a SimplerResolver that always has a null document.  
PowerApps has its own resolver with a non-nulldocument.

Reply:
+1 , any references to _txb.Document should have null checks

Reply:
I have used search to add couple of more null checks. Have no idea whether I covered all of them or whether they are correct.